mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.kt

```diff
@@ -965,6 +989,9 @@ class MainActivity : AbstractBaseActivity(), ConnectionFactory.UpdateListener {
         }
 
         controller.indicateServerCommunicationFailure(message)
+        scheduleRetry {
+            retryServerPropertyQuery()
+        }
```

Comment:
I would suggest to only selectively retry here. I don't think retrying on HTTP 4xx return codes provides any benefit.

Reply:
IMO a retry doesn't hurt either and there are some 4XX errors that can be "fixed" by waiting, e.g. 429 (Too many requests), 423 (Locked) or 408 (Request Timeout). (Not that I have ever seen these codes in the wild).